src/main/java/com/google/sps/servlets/EnterQueue.java

```diff
@@ -76,13 +75,29 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       FirebaseToken decodedToken = authInstance.verifyIdToken(idToken);
       String userID = decodedToken.getUid();
 
-      if (request.getParameter("enterTA") == null) {
+      // verify student status
+      UserRecord userRecord = authInstance.getUser(userID);
+      String userEmail = userRecord.getEmail();
```

Comment:
Why not use `decodedToken.getEmail()` here? That way you don't need to get the user record.